internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +99,71 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		errored := false
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		m.updateWithStartedCondition(ctx)
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// TODO: add support for additional decommissioning target types
+
+		if !errored {
+			m.updateWithCompletedCondition(ctx)
+			m.log.Warn("No errors during decommissioning, updated Condition to decommissioning completed")
+		}
+
+		// after this point the device will no longer be able to communicate with the management service
+		m.log.Warn("Preparing to wipe agent certificate and keys and reboot")
+		return m.wipeAndReboot(ctx)
+	}
 	return nil
 }
 
+func (m *LifecycleManager) updateWithStartedCondition(ctx context.Context) {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateStarted),
+		Message: "The device has started decommissioning",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+	}
+}
+
+func (m *LifecycleManager) updateWithCompletedCondition(ctx context.Context) {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateComplete),
+		Message: "The device has completed decommissioning and will wipe its management certificate",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+	}
+}
+
+// point of no return - wipes management cert and keys
+func (m *LifecycleManager) wipeAndReboot(ctx context.Context) error {
+	err := m.deviceReadWriter.RemoveFile(m.managementCertPath)
+	if err != nil {
+		m.log.Errorf("Unable to delete mcert: %v", err)
+		return err
+	}
+
+	m.enrollmentUIEndpoint = ""
```

Comment:
What are we wiping here and what is the purpose? The wipe is only from memory which will not persist after reboot

Reply:
Makes sense

Reply:
can we add a note

Reply:
Yep done